proto/frequenz/api/dispatch/dispatch.proto

```diff
@@ -116,31 +116,26 @@ message DispatchComponentIDs {
   repeated uint64 component_ids = 1;
 }
 
-// Parameters for filtering the dispatch list
-message DispatchFilter {
-  // Filter by dispatch ID
-  repeated uint64 ids = 1;
-
-  // Filter by microgrid ID
-  repeated uint64 microgrid_ids = 2;
+// Message for listing dispatches for a given microgrid, and an optional filter
+message DispatchListRequest {
+  // The microgrid ID
+  uint64 microgrid_id = 1;
 
-  // Filter by dispatch type
-  repeated string types = 3;
+  // Additional filter parameters
+  DispatchFilter filter = 2;
```

Comment:
It feels weird to have the these in a dedicated `DispatchFilter` message. These have the same filtering effect as the `microgrid_id`.
Is there a reason to design it specifically like this? One easy alternative could be just put the members of the `DispatchFilter` here, which will additionally also help to get rid of a step involving optionals.

Reply:
See Tom's comment here: https://github.com/frequenz-floss/frequenz-api-dispatch/issues/27#issuecomment-1650231924

The idea is that the microgrid ID is semantically different from the filter, so that listing dispatches always requires you to specify which MID it's for, and then you can additionally filter those with the optional `filter` field.